### PR TITLE
Fail run upon error

### LINKED ISSUE #25: Validator can fail without returning with a non-zero exit code
I've been having some trouble at ml-evs/odbx.science#15 with the validator action failing to install optimade (for whatever reason, probably incorrect version number), but then failing to return a non-zero exit code to actions (so the checks all pass). In my particular case, the failure occurs when importing optimade, after a failed install from pip (which itself should return a non-zero exit code...). I guess something needs to be done to return on the first non-zero exit code?

### PR DESCRIPTION
**Pull request by CasperWA**
Fixes #25 

`set -e` has been added to `entrypoint.sh`, so that any non-zero return value will stop the script.
Furthermore, an extra test for an invalid version value has been added.

### CODE REVIEW

**Comment by CasperWA** [comment]
I have successfully added a CI job that is meant to fail, and only succeeds if it fails.
In more detail, this action is called with an invalid version, and it should therefore fail with an exit code other than 0. If this is _not_ the case, the CI job will fail, otherwise it will succeed.

**Comment by CasperWA** [comment]
This PR is currently blocking for other PRs, since the added CI job has been made a requirement for merging into `master`.

**Comment by CasperWA** [comment]
> This PR is currently blocking for other PRs, since the added CI job has been made a requirement for merging into `master`.

This is no longer true, however, upon merging this PR, the branch protection should be updated to include the new CI job.

**Comment by CasperWA** [comment]
@ml-evs this is ready for review whenever :)

File: .github/workflows/validator_action.yml [hunk 1]
```diff
@@ -110,3 +110,32 @@ jobs:
...
+        uses: ./
+        with:
+          port: 3213
+          path: /v1
+          validator version: "0.0.0"
+
+      - name: This runs ONLY if 'Run action' doesn't fail as expected
+        if: steps.action.outcome != 'failure' || steps.action.conclusion != 'success'
+        run: |
+          echo "Outcome: ${{ steps.action.outcome }} (not 'failure'), Conclusion: ${{ steps.action.conclusion }} (not 'success')"
```
Comment (ml-evs) [inline .github/workflows/validator_action.yml:140]: Nice!

**Review by ml-evs — APPROVED** [review]
LGTM @CasperWA, thanks.

**Pull request merged**